grpc/wrappers.go

```diff
@@ -583,3 +558,865 @@ func (ras *RegistrationAuthorityServerWrapper) AdministrativelyRevokeCertificate
 	}
 	return &corepb.Empty{}, nil
 }
+
+// StorageAuthorityClientWrapper is the gRPC version of a core.StorageAuthority client
+type StorageAuthorityClientWrapper struct {
+	inner   sapb.StorageAuthorityClient
+	timeout time.Duration
+}
+
+func NewStorageAuthorityClient(inner sapb.StorageAuthorityClient, timeout time.Duration) *StorageAuthorityClientWrapper {
+	return &StorageAuthorityClientWrapper{inner, timeout}
+}
+
+func (sac StorageAuthorityClientWrapper) GetRegistration(ctx context.Context, regID int64) (core.Registration, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	response, err := sac.inner.GetRegistration(localCtx, &sapb.RegistrationID{Id: &regID})
+	if err != nil {
+		return core.Registration{}, unwrapError(err)
+	}
+
+	if response == nil || !registrationValid(response) {
+		return core.Registration{}, errIncompleteResponse
+	}
+
+	return pbToRegistration(response)
+}
+
+func (sac StorageAuthorityClientWrapper) GetRegistrationByKey(ctx context.Context, key *jose.JsonWebKey) (core.Registration, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	keyBytes, err := key.MarshalJSON()
+	if err != nil {
+		return core.Registration{}, err
+	}
+
+	response, err := sac.inner.GetRegistrationByKey(localCtx, &sapb.JsonWebKey{Jwk: keyBytes})
+	if err != nil {
+		return core.Registration{}, unwrapError(err)
+	}
+
+	if response == nil || !registrationValid(response) {
+		return core.Registration{}, errIncompleteResponse
+	}
+
+	return pbToRegistration(response)
+}
+
+func (sac StorageAuthorityClientWrapper) GetAuthorization(ctx context.Context, authID string) (core.Authorization, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	response, err := sac.inner.GetAuthorization(localCtx, &sapb.AuthorizationID{Id: &authID})
+	if err != nil {
+		return core.Authorization{}, unwrapError(err)
+	}
+
+	if response == nil || !authorizationValid(response) {
+		return core.Authorization{}, errIncompleteResponse
+	}
+
+	return pbToAuthz(response)
+}
+
+func (sac StorageAuthorityClientWrapper) GetValidAuthorizations(ctx context.Context, regID int64, domains []string, now time.Time) (map[string]*core.Authorization, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	nowUnix := now.UnixNano()
+
+	response, err := sac.inner.GetValidAuthorizations(localCtx, &sapb.GetValidAuthorizationsRequest{
+		RegistrationID: &regID,
+		Domains:        domains,
+		Now:            &nowUnix,
+	})
+	if err != nil {
+		return nil, unwrapError(err)
+	}
+
+	if response == nil {
```

Comment:
Is there a known case where response can be nil when err is nil? Or is this just defensive code?